### PR TITLE
use ICU

### PR DESCRIPTION
**Pull request by mr-martian**
ICU
- convert all `std::wstring`s and related types to `UString`
- use `lttoolbox/input_file.h` for reading UTF-8 input with nulls
- use `UFILE*` for output

efficiency, cleanliness and code style
- move constant initializers to header file
- store references to special transducer symbols to save `alphabet` lookups
- use range-for loops when possible
- prefer `std::vector` to `std::list`
- prefer `str.empty()` to `str == ""`
- drop old, unused file in `src/`

helper functions
- `XMLParseUtil` has more specialized functions now
- `XMLParseUtil` is now in lttoolbox, so drop apertium dependency

### CODE REVIEW

File: src/lsx_compiler.cc [hunk 1]
```diff
@@ -29,14 +29,15 @@ using namespace std;
...
+UString const Compiler::COMPILER_ANYTAG_ELEM        = "t"_u;
+UString const Compiler::COMPILER_ANYCHAR_ELEM       = "w"_u;
+UString const Compiler::COMPILER_WB_ELEM            = "j"_u;
 
 Compiler::Compiler() :
 reader(0),
 verbose(false),
-first_element(false)
+first_element(false),
+any_tag(0), any_char(0), word_boundary(0)
```
Comment (TinoDidriksen) [inline src/lsx_compiler.cc:40]: These can go in the class body.

**Comment by TinoDidriksen** [comment]
LGTM

File: src/lsx_compiler.cc (hunk 1 above)
Comment (mr-martian) [reply src/lsx_compiler.cc:40]: not entirely sure what you mean by that

Comment (TinoDidriksen) [reply src/lsx_compiler.cc:40]: Class member initializers can go into the class body as of C++11, so you can just write `int32_t any_tag = 0;` or `int32_t any_tag{0};` in the class and then don't have to worry about constructor init.

**Pull request merged**